### PR TITLE
Vehicle Object Load

### PR DESCRIPTION
**Pull request by duncanspumpkin**
~~Wip~~ Done

### CODE REVIEW

File: src/OpenLoco/src/Objects/VehicleObject.cpp [hunk 1]
```diff
@@ -307,29 +309,284 @@ namespace OpenLoco
...
+            if (!bodySprite.hasFlags(BodySpriteFlags::hasSprites))
             {
-                std::copy(reinterpret_cast<ObjectHeader*>(depObjs), reinterpret_cast<ObjectHeader*>(depObjs) + dependencies->required.size(), dependencies->required.data());
-                depObjs += sizeof(ObjectHeader) * dependencies->required.size();
+                continue;
             }
-            dependencies->willLoad.resize(*depObjs++);
-            if (!dependencies->willLoad.empty())
+            bodySprite.flatImageId = offset + imgRes.imageOffset;
+            bodySprite.var_0B = getNumRotationSpriteTypesFlat(bodySprite.numFlatRotationFrames);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/VehicleObject.cpp:514]: Can we name `var_0B` at this point?

File: src/OpenLoco/src/Objects/VehicleObject.cpp [hunk 2]
```diff
@@ -307,29 +309,284 @@ namespace OpenLoco
...
+            offset += numFlatFrames / (bodySprite.hasFlags(BodySpriteFlags::rotationalSymmetry) ? 2 : 1);
+
+            if (bodySprite.hasFlags(BodySpriteFlags::hasGentleSprites))
             {
-                std::copy(reinterpret_cast<ObjectHeader*>(depObjs), reinterpret_cast<ObjectHeader*>(depObjs) + dependencies->willLoad.size(), dependencies->willLoad.data());
+                bodySprite.gentleImageId = offset + imgRes.imageOffset;
+                const auto numGentleFrames = bodySprite.numFramesPerRotation * 8;
+                offset += numGentleFrames / (bodySprite.hasFlags(BodySpriteFlags::rotationalSymmetry) ? 2 : 1);
+
+                bodySprite.var_0C = getNumRotationSpriteTypesSloped(bodySprite.numSlopedRotationFrames);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/VehicleObject.cpp:526]: Can we name `var_0C` at this point?

File: src/OpenLoco/src/Objects/VehicleObject.cpp [hunk 3]
```diff
@@ -307,29 +309,284 @@ namespace OpenLoco
...
+                bodySprite.unkImageId = offset + imgRes.imageOffset;
+                const auto numUnkFrames = bodySprite.numFlatRotationFrames * 3;
+                offset += numUnkFrames / (bodySprite.hasFlags(BodySpriteFlags::rotationalSymmetry) ? 2 : 1);
+            }
+
+            const auto numImages = imgRes.imageOffset + offset - bodySprite.flatImageId;
+            const auto extents = Gfx::getImagesMaxExtent(ImageId(bodySprite.flatImageId), numImages);
+            bodySprite.var_08 = extents.width;
+            bodySprite.var_09 = extents.heightNegative;
+            bodySprite.var_0A = extents.heightPositive;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/VehicleObject.cpp:552]: Can we name these yet?

File: src/OpenLoco/src/Objects/VehicleObject.cpp [hunk 4]
```diff
@@ -307,29 +309,284 @@ namespace OpenLoco
...
+                    const auto numSteepFrames = bogieSprite.numRollSprites * 64;
+                    offset += numSteepFrames / (bogieSprite.hasFlags(BogieSpriteFlags::rotationalSymmetry) ? 2 : 1);
+                }
+            }
+
+            const auto numImages = imgRes.imageOffset + offset - bogieSprite.flatImageIds;
+            const auto extents = Gfx::getImagesMaxExtent(ImageId(bogieSprite.flatImageIds), numImages);
+            bogieSprite.var_02 = extents.width;
+            bogieSprite.var_03 = extents.heightNegative;
+            bogieSprite.var_04 = extents.heightPositive;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/VehicleObject.cpp:585]: Can we name these yet?

**Comment by duncanspumpkin** [comment]
Note: Should really look at reworking how the sprite rotational symmetry works. I don't like all those divides and they aren't really required.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**